Drv/LinuxSpiDriver/LinuxSpiDriverComponentImpl.cpp

```diff
@@ -97,9 +98,27 @@ namespace Drv {
 
         // Configure:
         /*
-         * SPI Mode 0
+         * SPI Mode 0, 1, 2, 3
          */
-        U8 mode = SPI_MODE_0; // Mode 0 (CPOL = 0, CPHA = 0)
+
+        U8 mode; // Mode Select (CPOL = 0/1, CPHA = 0/1)
+        switch(spiMode) {
+            case SpiMode::SPI_MODE_0:
+                mode = SPI_MODE_0;
+                break;
+            case SpiMode::SPI_MODE_1:
+                mode = SPI_MODE_1;
+                break;
+            case SpiMode::SPI_MODE_2:
+                mode = SPI_MODE_2;
+            case SpiMode::SPI_MODE_3:
+                mode = SPI_MODE_3;
+            default:
+                //Assert if the device SPI Mode is not in the correct range
+                FW_ASSERT(spiMode >= SpiMode::SPI_MODE_0 || spiMode <= SpiMode::SPI_MODE_3);                
```

Comment:
You should just assert zero in this case.

```
FW_ASSERT(0, spiMode);
```

This will force-fail the assert and print the invalid mode.

Reply:
@LeStarch Made the fix